src/main/java/games/strategy/util/Util.java

```diff
@@ -38,19 +40,10 @@
    * @return true if some element in c1 is in c2
    */
   public static <T> boolean someIntersect(final Collection<T> c1, final Collection<T> c2) {
```

Comment:
I think the implementation can be reduced to just:

```java
return c1.stream().anyMatch(c2::contains);
```

This appears to properly handle the empty cases.  The following test passed for all three implementations (current, proposed, and proposed without guard clause):

```java
@Test
public void testSomeIntersect() {
  // one or both empty
  assertThat(Util.someIntersect(Arrays.asList(), Arrays.asList()), is(false));
  assertThat(Util.someIntersect(Arrays.asList(), Arrays.asList("a")), is(false));
  assertThat(Util.someIntersect(Arrays.asList("a"), Arrays.asList()), is(false));

  // intersection
  assertThat(Util.someIntersect(Arrays.asList("a"), Arrays.asList("a")), is(true));
  assertThat(Util.someIntersect(Arrays.asList("a", "b"), Arrays.asList("a")), is(true));
  assertThat(Util.someIntersect(Arrays.asList("a"), Arrays.asList("a", "b")), is(true));
  assertThat(Util.someIntersect(Arrays.asList("a", "b"), Arrays.asList("b", "c")), is(true));

  // no intersection
  assertThat(Util.someIntersect(Arrays.asList("a", "b"), Arrays.asList("c", "d")), is(false));
}
```